### PR TITLE
Fixed issue with csv file with excel content-type when inferring dataset.

### PR DESCRIPTION
**Pull request by serge-gaia**
Rewrite the file format detection in FileReader.

See
https://decbugs.com/view.php?id=6871

### CODE REVIEW

File: biosys/apps/main/api/uploaders.py [hunk 1]
```diff
@@ -63,17 +63,40 @@ class FileReader(object):
...
+        :return:
+        """
+        if not hasattr(uploaded_file, 'name') or not hasattr(uploaded_file, 'content_type'):
+            raise Exception("The given file is not a django managed uploaded file: {}".format(uploaded_file))
+        # check extension first
+        # Note: on Windows the content-type for a csv file can be sent with an excel mime type.
+        file_name = uploaded_file.name
+        content_type = uploaded_file.content_type
+        extension = path.splitext(file_name)[1].lower()
+        result = FileReader.NOT_SUPPORTED_FORMAT
```
Comment (tony-gaia) [inline biosys/apps/main/api/uploaders.py]: for clarity i'd put this line in an else statement below

File: biosys/apps/main/tests/api/test_schema_inference.py [hunk 2]
```diff
@@ -260,6 +262,90 @@ def test_mix_types_infer_most_plausible(self):
...
+        """
+        Often on Windows a csv file comes with an excel content-type (e.g: 'application/vnd.ms-excel')
+        Test that we handle the case.
+        """
+        # In order to hack the Content-Type of the multipart form data we need to use the APIRequestFactory and work
+        # with the view directly. Can't use the classic API client.
+        from rest_framework.authtoken.models import Token
+        from rest_framework.test import APIRequestFactory
+        from main.api.views import InferDatasetView
+        from rest_framework.test import force_authenticate
```
Comment (tony-gaia) [inline biosys/apps/main/tests/api/test_schema_inference.py:275]: Do theses statements need to be within the function?

Comment (serge-gaia) [reply biosys/apps/main/tests/api/test_schema_inference.py:275]: Nope but they are very specific to this test. So I thought of localising them in the function.

File: biosys/apps/main/api/uploaders.py (hunk 1 above)
Comment (serge-gaia) [reply biosys/apps/main/api/uploaders.py]: will do

Comment (serge-gaia) [reply biosys/apps/main/api/uploaders.py]: I did it like that first.

File: biosys/apps/main/tests/api/test_schema_inference.py (hunk 2 above)
Comment (serge-gaia) [reply biosys/apps/main/tests/api/test_schema_inference.py:275]: with comment

Comment (tony-gaia) [reply biosys/apps/main/tests/api/test_schema_inference.py:275]: Pep 8 recommends putting all imports at the top of the module.

https://www.python.org/dev/peps/pep-0008/#imports

Only exception is if you have circular imports

Comment (serge-gaia) [reply biosys/apps/main/tests/api/test_schema_inference.py:275]: I know, I know, Mr. Pep8. I'll fix that.

Comment (tony-gaia) [reply biosys/apps/main/tests/api/test_schema_inference.py:275]: You so could have said Dr Pep8 (like Dr Pepper).

**Pull request merged**